modoboa/extensions/stats/management/commands/logparser.py

```diff
@@ -297,6 +297,14 @@ def _parse_line(self, line):
         if not m:
             return
         host, prog, pid, log = m.groups()
+        m = re.search("(INFECTED|SPAMMY) .* <[^>]+> -> <[^@]+@([^>]+)>.*", log)
```

Comment:
The regular expression catches the type virus (INFECTED) or spam (SPAMMY) in group 1, group 2 contains the domain. The if statements on lines 303 and 305 identify it its spam or a virus and increment the relevant counter.